kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/JdbcAuthenticationProviderImpl.scala

```diff
@@ -0,0 +1,232 @@
+   * @throws AuthenticationException When a user is found to be invalid by the implementation
+   */
+  @throws[AuthenticationException]
+  override def authenticate(user: String, password: String): Unit = {
+    if (StringUtils.isBlank(user)) {
+      throw new AuthenticationException(s"Error validating, user is null" +
+        s" or contains blank space")
+    }
+
+    if (StringUtils.isBlank(password)) {
+      throw new AuthenticationException(s"Error validating, password is null" +
+        s" or contains blank space")
+    }
+
+    checkConfigs
+
+    loadJdbcDriverClass
+
+    var connection: Connection = null
+    var queryStatement: PreparedStatement = null
+
+    try {
+      connection = DriverManager.getConnection(dbUrl.get, dbUserName.orNull, dbPassword.orNull)
+
+      queryStatement = getAndPrepareStatement(connection, user, password)
+
+      val resultSet = queryStatement.executeQuery()
+
+      if (resultSet == null || !resultSet.next()) {
+        // Auth failed
+        throw new AuthenticationException(s"Password does not match or no such user. user:" +
+          s" $user , password length: ${password.length}")
+      }
+
+      // Auth passed
+
+    } catch {
+      case e: AuthenticationException =>
+        throw e
+      case e: Exception =>
+        error("Cannot get user info", e);
+        throw e
+    } finally {
+      closeDbConnection(connection, queryStatement)
+    }
+  }
+
+  private def checkConfigs: Unit = {
+    def configLog(config: String, value: String): String = s"JDBCAuthConfig: $config = '$value'"
+
+    debug(configLog("Driver Class", dbDriver.orNull))
+    debug(configLog("JDBC URL", dbUrl.orNull))
+    debug(configLog("Database Username", dbUserName.orNull))
+    debug(configLog("Database Password", dbPassword.orNull))
+    debug(configLog("Query SQL", querySql.orNull))
+
+    // Check if JDBC parameters valid
+    if (dbDriver.isEmpty) {
+      throw new IllegalArgumentException("JDBC driver class is not configured.")
+    }
+
+    if (dbUrl.isEmpty) {
+      throw new IllegalArgumentException("JDBC url is not configured")
+    }
+
+    if (dbUserName.isEmpty || dbPassword.isEmpty) {
+      throw new IllegalArgumentException("JDBC username or password is not configured")
+    }
+
+    // Check Query SQL
+    if (querySql.isEmpty) {
+      throw new IllegalArgumentException("Query SQL is not configured")
+    }
+    val querySqlInLowerCase = querySql.get.trim.toLowerCase
+    if (!querySqlInLowerCase.startsWith("select")) { // allow select query sql only
+      throw new IllegalArgumentException("Query SQL must start with \"SELECT\"");
+    }
+    if (!querySqlInLowerCase.contains("where")) {
+      warn("Query SQL does not contains \"WHERE\" keyword");
+    }
+    if (!querySqlInLowerCase.contains("${username}")) {
+      warn("Query SQL does not contains \"${username}\" placeholder");
+    }
+  }
+
+  private def loadJdbcDriverClass: Unit = {
```

Comment:
HikariDataSource can be used here to simplify JDBC code
`org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStore`

Reply:
I didn't use HikariCP here for minimum depedency changes, since I have noticed kyuubi-common module does not depend on HikariCP (while kyuubi-server does).

In issue https://github.com/apache/incubator-kyuubi/issues/3222, the pooling feature was mentioned to provide.

Or let me follow your idea and change to use HikariDataSource here.

Reply:
It's ok to add HikariCP in kyuubi-common module.
Hikari provides both pooled `HikariDataSource` and unpooled `DriverDataSource` implementations of `DataSource`, which can help us to narrow code differences.

Reply:
migrated to use HikariDataSource.